src/import.rs

```diff
@@ -203,5 +201,32 @@ fn import_impl(path: &Path) -> Result<Import> {
 pub fn import<P>(path: P) -> Result<Import>
     where P: AsRef<Path>
 {
-    import_impl(path.as_ref())
+    let path = path.as_ref();
+    let base = path.parent().unwrap_or(Path::new("./"));
+    let file = fs::File::open(path).map_err(Error::Io)?;
+    let reader = io::BufReader::new(file);
+    import_impl(Gltf::from_reader(reader)?, Some(base))
```

Comment:
We need to keep this similiar to before or else the compiler will generate lots of code, once for every `P` type. I suggest we add another intermediate function:

```rust
// New function (name could be improved!)
fn import_path(path: &Path) -> Result<Import> {
    ...
    import_impl(Gltf::from_reader(reader)?, Some(base))
}

// As you have
fn import_impl(Gltf { document, blob }: Gltf, base: Option<&Path>) -> Result<Import> {
    ...
}

// As before
pub fn import<P>(path: P) -> Result<Import>
    where P: AsRef<Path>,
{
    import_path(path.as_ref())
}
```

We could also make the slice parameter more generic:
```rust
fn import_slice_impl(slice: &[u8]) -> Result<Import> {
    import_impl(Gltf::from_slice(slice)?, None)
}

pub fn import_slice<S>(slice: S) -> Result<Import>
    where S: AsRef<[u8]>,
{
    import_slice_impl(slice.as_ref())
}
```

Reply:
Done I think 👌